src/main/scala/beam/agentsim/events/handling/GoogleTravelTimeEstimationEntry.scala

```diff
@@ -0,0 +1,150 @@
+      entry.originLng,
+      entry.destLat,
+      entry.destLng,
+      entry.simTravelTime,
+      entry.googleTravelTime,
+      entry.googleTravelTimeWithTraffic,
+      entry.euclideanDistanceInMeters,
+      entry.legLength,
+      entry.googleDistance
+    )
+  }
+
+  def fromJulMap(map: java.util.Map[String, String]): GoogleTravelTimeEstimationEntry = {
+    GoogleTravelTimeEstimationEntry(
+      requestId = map.get("requestId"),
+      vehicleId = Id.createVehicleId(map.get("vehicleId")),
+      vehicleType = map.get("vehicleType"),
+      departureTime = map.get("departureTime").toInt,
+      originLat = map.get("originLat").toDouble,
+      originLng = map.get("originLng").toDouble,
+      destLat = map.get("destLat").toDouble,
+      destLng = map.get("destLng").toDouble,
+      simTravelTime = map.get("simTravelTime").toInt,
+      googleTravelTime = map.get("googleTravelTime").toInt,
+      googleTravelTimeWithTraffic = Option(map.get("googleTravelTimeWithTraffic")).map(_.toInt),
+      euclideanDistanceInMeters = map.get("euclideanDistanceInMeters").toDouble,
+      legLength = map.get("legLength").toDouble,
+      googleDistance = map.get("googleDistance").toInt
+    )
+  }
+
+  def fromPTE(
+    pte: PathTraversalEvent,
+    requestId: String,
+    googleTravelTime: Long,
+    googleTravelTimeWithTraffic: Option[Long],
+    googleDistance: Long,
+    geoUtils: GeoUtils
+  ): GoogleTravelTimeEstimationEntry = {
+    GoogleTravelTimeEstimationEntry(
+      requestId = requestId,
+      vehicleId = pte.vehicleId,
+      vehicleType = pte.vehicleType,
+      departureTime = pte.departureTime,
+      originLat = pte.startY,
+      originLng = pte.startX,
+      destLat = pte.endY,
+      destLng = pte.endX,
+      simTravelTime = pte.arrivalTime - pte.departureTime,
+      googleTravelTime = googleTravelTime,
+      googleTravelTimeWithTraffic = googleTravelTimeWithTraffic,
+      euclideanDistanceInMeters = geoUtils.distLatLon2Meters(
+        new Coord(pte.startX, pte.startY),
+        new Coord(pte.endX, pte.endY)
+      ),
+      legLength = pte.legLength,
+      googleDistance = googleDistance
+    )
+  }
+
+  object Csv {
+
+    val headers = Vector(
+      "requestId",
+      "vehicleId",
+      "vehicleType",
+      "departureTime",
+      "originLat",
+      "originLng",
+      "destLat",
+      "destLng",
+      "simTravelTime",
+      "googleTravelTime",
+      "googleTravelTimeWithTraffic",
+      "euclideanDistanceInMeters",
+      "legLength",
+      "googleDistance"
+    )
+
+    def writeEntries(entries: Seq[GoogleTravelTimeEstimationEntry], filePath: String): Unit = {
+      using(new CsvWriter(filePath, headers)) { csvWriter =>
+        entries.map(GoogleTravelTimeEstimationEntry.toRow).foreach(csvWriter.writeRow)
+      }
+    }
+
+    def parseEntries(text: String): Seq[GoogleTravelTimeEstimationEntry] = {
+      val csvReader = new CsvMapReader(new StringReader(text), CsvPreference.STANDARD_PREFERENCE)
```

Comment:
This should be in a using (maybe the string reader also, but the csvmapreader probably handles it in it's close - or should ;) )